### DISCUSSION TITLE
221020 Meeting ideas & protocol

### DISCUSSION THREAD

**Discussion by ehhartmann** [graeter-group/kimmdy#75] [Ideas]
# Introduction
- what are people working on (not necessarily related to KIMMDY)?
-

# New Features since March
- reaction plugin architecture
- dummy HAT reaction
- HAT NN reaction
- changemanager for HAT reaction
- changemanager tests, some general tests
- file history, easy file handling 
- more general mdmanager
- relaxation by short MD
- automated documentation

# Feature debate
- mdmanager: which command line options are always used, which optional?
- relaxation: parallel existence of multiple schemes desirable? maintaining detailed balance and physical behavior
- HAT NN reaction: if we decide for a reaction, should the geometry change end at exactly this position?
- MC theory: how do we get reaction rates for competing reactions? attempt frequency, multiple snapshots

# General
- when to release KIMMDY v2?
- rename modified force field and remove deficient versions
-

**Comment by jmbuhr** [comment]
- more general changemanger not reliant on hydrogens i.e. safe handing of topology changes

**Comment by jmbuhr** [comment]
- Checkpoint files and restarts for kimmdy (and internal md runs?)

**Comment by jmbuhr** [comment]
![image](https://user-images.githubusercontent.com/17450586/196953605-ad1cad6d-e654-4b97-bded-5b006830da47.png)

**Comment by KRiedmiller** [comment]
# Protocol

* How to get prefactor? QM/MM? pure QM?
* Checkpoints for testing of individual parts
* cmdline for gmx:
	* specify everything all the time
	* no defname
	* files not found are no issue, as long all necessary information are somewhere
* MC w/ rejection:
	* cutoff in time domain
* How to mix reactions?
	* option 1: Only do one kind at a time
	* option 2: Competing rates need to be evaluated at the same time (as we currently do)
	* Competing rates can have diff prefactors
* maybe recalculate prefactor for homolysis in QMMM together with HAT QMMM
* KIMMDY Checkpoints:
	* save task queue maybe as pickle
	* gmx restarts automatically if chk is found
	* option: save to new config, problem: objects in queue
	* save current state (`gmx running`, or `predicting rates`..)
	* save runmanager.latest_files
* How to jump:
	* morse does the job
	* teleport H to ideal position, copy v of connected atom
	* lambda dynamics of bonds, can be reversible
